src/ahocode.py

```diff
@@ -0,0 +1,341 @@
+AHOCORASICK = 2
+
+EMPTY = 0
+
+KEY_SEQUENCE = 200
+KEY_STRING = 100
+
+MATCH_AT_LEAST_PREFIX = 2
+
+MATCH_AT_MOST_PREFIX = 1
+
+MATCH_EXACT_LENGTH = 0
+
+STORE_ANY = 30
+STORE_INTS = 10
+STORE_LENGTH = 20
+
+TRIE = 1
+
+unicode = 1
+
+
+class State(object):
+    __slots__ = ['identifier', 'symbol', 'success', 'transitions', 'parent',
+                 'matched_keyword', 'longest_strict_suffix', 'value']
+
+    def __init__(self, identifier, symbol=None, parent=None, success=False, value=None):
+        self.symbol = symbol
+        self.identifier = identifier
+        self.transitions = {}
+        self.parent = parent
+        self.success = success
+        self.matched_keyword = None
+        self.value = value
+        self.longest_strict_suffix = None
+
+    def __str__(self):
+        transitions_as_string = ','.join(
+            ['{0} -> {1}'.format(key, value.identifier) for key, value in self.transitions.items()])

```

Comment:
I don't think you need to have your list comprehension in a list if you're passing it directly into `.join()`.
```suggestion
            '{0} -> {1}'.format(key, value.identifier) for key, value in self.transitions.items())

```